src/main/re_pressed/impl.cljs

```diff
@@ -146,78 +145,77 @@
-                                   "TEXTAREA"} tag-name)
-                ;; --
-
-                hit-key {:altKey   (.-altKey e)
-                         :ctrlKey  (.-ctrlKey e)
-                         :metaKey  (.-metaKey e)
-                         :shiftKey (.-shiftKey e)
-                         :which    (.-which e)}
-
-                always-listen-keys @(rf/subscribe [(ns-keyword "-always-listen-keys")])
-                always-listen?     (some #(is-key? hit-key %) always-listen-keys)]
-
-            (when (or (and (not modifier-key?)
-                           (not entering-input?))
-                      always-listen?)
-              (rf/dispatch-sync [(ns-keyword "-set-key") hit-key])
-
-              (let [recent-keys @(rf/subscribe [(ns-keyword "-keys")])
-                    event-keys  @(rf/subscribe [(ns-keyword "-event-keys")])
-                    clear-keys  @(rf/subscribe [(ns-keyword "-clear-keys")])
-
-                    recent-key (last recent-keys)
-
-                    is-key-sequence? (fn [key-maps]
-                                       (every? true?
-                                               (mapv
-                                                #(is-key? %1 %2)
-                                                (concat (reverse recent-keys)
-                                                        (repeat max-record {}))
-                                                (reverse key-maps))))
-
-                    check-events (doall
-                                  (for [[trigger-event
-                                         & ks] event-keys]
-                                    (let [triggered? (some is-key-sequence? ks)]
-                                      (vector trigger-event triggered?))))
-
-                    [triggered-event
-                     event?] (some->> check-events
-                                      (filter #(true? (second %)))
-                                      first)
-
-                    clear? (some is-key-sequence? clear-keys)]
-
-                (when (= "keydown" event-type)
-                  (let [prevent-default-keys @(rf/subscribe [(ns-keyword "-prevent-default-keys")])
-                        prevent-default?     (some #(is-key? recent-key %) prevent-default-keys)]
-                    (when prevent-default?
-                      (rf/dispatch-sync [(ns-keyword "-prevent-default-keys")
-                                         e]))))
-
-                (cond
-                  clear?
-                  (rf/dispatch-sync [(ns-keyword "-clear-keys")])
-
-                  (and event?)
-                  (rf/dispatch-sync (conj triggered-event
-                                          e
-                                          recent-keys))
-
-                  :else nil))))))))
+   (ev/listen
+      js/document
+      event-type
+      (fn [e]
+        (let [ns-keyword (->ns-keyword event-type)
+
+              e-key         (.-which e)
```

Comment:
@heyarne this should be .-keyCode instead of .-which

Reply:
True, I didn't change it here. But from what I can tell the logic for detecting modifiers is flawed then: `(.-keyCode e)` never holds the code for Ctrl / Shift / Alt. What were you trying to detect here?

Reply:
@heyarne in the original code, (.-which e) is used specifically to make sure that the key that was just pressed *isn't* a modifier key. The code is filtering out modifier keys and not adding them to the stack of keypresses that a user is matching against. However, a user still get to know if a modifier key is pressed **with** a key by way of hit-key.

This is so if you are checking that sequence of keys like the following:
Ctrl+A
Shift+B
Alt+C

What is recorded is a vector if three items, instead of 6 items where the modifier keys get recorded individually.

What we don't want is to record this:
Ctrl
A
Shift
B
Alt
C

And have to figure out that the modifier is meant for the next key.

Reply:
@heyarne In the original code if you replace the next line with: `modifier-key? false` and run the demo application, press Ctrl by itself, you can see the following output to the screen:

```clojure
{:altKey false,
 :ctrlKey true, 
 :metaKey false, 
 :shiftKey false, 
 :which 17} ;; <-- the int for the Ctrl key is 17 ... the modifier-key? is filtering this out effectively
```

However, if you leave the original code as it was and press the ctrl key alone, nothing is recorded (the desired behavior).